bp_common/syn/tcl/vivado_synth.tcl

```diff
@@ -23,18 +24,28 @@ foreach x $f {
       set temp [string trimleft $x $trimchars]
       set expanded [subst $temp]
       lappend dir_list $expanded
+    } elseif {[string match "*bsg_mem_1rw_sync_mask_write_bit.v" $x]} {
+      set replace_hard "$BASEJUMP_STL_DIR/hard/ultrascale_plus/bsg_mem/bsg_mem_1rw_sync_mask_write_bit.v"
+      set expanded [subst $replace_hard]
+      lappend flist $expanded
+      puts $expanded
     } else {
       set expanded [subst $x]
       lappend flist $expanded
     }
   }
 }
 
+puts $flist
+
 set_part $PART
 read_verilog -sv $flist
 read_xdc design.xdc
 
-synth_design -top wrapper -part $PART -include_dirs $dir_list
+synth_design -top wrapper -part $PART -include_dirs $dir_list -flatten_hierarchy none
 report_utilization -file $REPORT_DIR/hier_util.rpt -hierarchical -hierarchical_percentages
 report_timing_summary -file $REPORT_DIR/timing.rpt
+write_checkpoint -force post_synthesis.dcp
```

Comment:
Do we need to write a checkpoint in order to do the next steps?

Reply:
No, a checkpoint isn't required.